src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/BuildAssetsUtils.cs

```diff
@@ -22,428 +22,693 @@ namespace NuGet.Commands
+            }
+        }
 
-                Debug.Assert((!request.Project.RestoreMetadata.CrossTargeting && (request.Project.TargetFrameworks.Count < 2)
-                    || (request.Project.RestoreMetadata.CrossTargeting)),
-                    "Invalid crosstargeting and framework count combination");
+        /// <summary>
+        /// Apply standard properties in a property group.
+        /// </summary>
+        public static void AddNuGetProperties(XDocument doc, IEnumerable<string> packageFolders, string repositoryRoot, RestoreOutputType outputType, bool success)
+        {
+            var projectStyle = "Unknown";
 
-                if (isCrossTargeting)
-                {
-                    // Find all global targets from buildCrossTargeting
-                    var crossTargetingAssets = GetTargetsAndPropsForCrossTargeting(
-                            targetGraphs,
-                            repositories,
-                            context,
-                            request,
-                            includeFlagGraphs);
-
-                    var crossProps = new MSBuildRestoreImportGroup();
-                    crossProps.Position = 0;
-                    crossProps.Conditions.Add(CrossTargetingCondition);
-                    crossProps.Imports.AddRange(crossTargetingAssets.Props);
-                    props.Add(crossProps);
-
-                    var crossTargets = new MSBuildRestoreImportGroup();
-                    crossTargets.Position = 0;
-                    crossTargets.Conditions.Add(CrossTargetingCondition);
-                    crossTargets.Imports.AddRange(crossTargetingAssets.Targets);
-                    targets.Add(crossTargets);
-                }
+            if (outputType == RestoreOutputType.NETCore)
+            {
+                projectStyle = "PackageReference";
+            }
+            else if (outputType == RestoreOutputType.UAP)
+            {
+                projectStyle = "ProjectJson";
+            }
 
-                // Find TFM specific assets from the build folder
-                foreach (var pair in buildAssetsByFramework)
-                {
-                    // There could be multiple string matches
-                    foreach (var match in GetMatchingFrameworkStrings(project, pair.Key))
-                    {
-                        var frameworkCondition = string.Format(CultureInfo.InvariantCulture, TargetFrameworkCondition, match);
+            doc.Root.AddFirst(
+                new XElement(Namespace + "PropertyGroup",
+                            new XAttribute("Condition", $" {ExcludeAllCondition} "),
+                            GetProperty("RestoreSuccess", success.ToString()),
+                            GetProperty("RestoreTool", "NuGet"),
+                            GetProperty("NuGetPackageRoot", ReplacePathsWithMacros(repositoryRoot)),
+                            GetProperty("NuGetPackageFolders", string.Join(";", packageFolders)),
```

Comment:
>packageFolders [](start = 80, length = 14)

Noted this is an absolute path. Although it points to the same location as `NuGetPackageRoot`: `C:\Users\alpanov\.nuget\packages\`.

Should it be also substituted with a macro?